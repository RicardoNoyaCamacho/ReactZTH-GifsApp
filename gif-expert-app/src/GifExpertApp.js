import React, { useState } from "react";

export const GifExpertApp = () => {
  //   const catergories = ["One Punch", "Samuray X", "Dragon Ball"];

  const [categories, setCategories] = useState(["One Punch","Samuray X","Dragon Ball"]);

  const handleAdd = () => {
    setCategories((cat) => ["One Piece", ...categories]);
  };

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <button onClick={handleAdd}>Agregar</button>
      <ol>
        {categories.map((cat) => {
          return <li key={cat}>{cat}</li>;
        })}
      </ol>
    </>
  );
};
