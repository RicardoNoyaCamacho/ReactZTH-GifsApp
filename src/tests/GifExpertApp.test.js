import React from "react";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas <GifExpertApp/>", () => {
  test("Debe mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar una lista de categorias", () => {
    const categories = ["Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
