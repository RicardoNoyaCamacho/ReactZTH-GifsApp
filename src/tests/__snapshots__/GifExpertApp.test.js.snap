// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={
        Array [
          "Dragon Ball",
        ]
      }
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas <GifExpertApp/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={Array []}
      key=""
    />
  </ol>
</Fragment>
`;
